core/src/main/java/org/elasticsearch/transport/TransportService.java

```diff
@@ -330,36 +330,35 @@ public DiscoveryNode connectToNodeAndHandshake(
     public DiscoveryNode connectToNodeAndHandshake(
             final DiscoveryNode node,
             final long handshakeTimeout,
-            final boolean checkClusterName) {
+            final boolean checkClusterName) throws IOException {
```

Comment:
I can remove it here but I think we will likely add it back in the future when we try to connect to foreign clusters. Lets trash it it's as simple to add back as it is to remove